index.bs

```diff
@@ -1804,6 +1804,96 @@ The [=remote end steps=] with |session| and <var ignore>command parameters</var>
 
 </div>
 
+#### The browser.createPartition Command #### {#command-browser-createPartition}
+
+The <dfn export for=commands>browser.createPartition</dfn> command creates a storage partition for a group of browsing contexts.
+
+<dl>
+   <dt>Command Type</dt>
+   <dd>
+      <pre class="cddl remote-cddl">
+      browser.Close = (
+        method: "browser.createPartition",
+        params: EmptyParams,
+      )
+      </pre>
+   </dd>
+   <dt>Return Type</dt>
+   <dd>
+    <pre class="cddl local-cddl">
+      browser.createPartitionResult = {
+        partition: browser.Partition
+      }
+    </pre>
+   </dd>
+</dl>
+
+<div algorithm="remote end steps for browser.createPartition">
+
+</div>
+
+#### The browser.closePartition Command #### {#command-browser-closePartition}
+
+The <dfn export for=commands>browser.closePartition</dfn> command closes the partition and all browsing contexts in it without running <code>beforeunload</code> handlers.
+
+<dl>
+   <dt>Command Type</dt>
+   <dd>
+      <pre class="cddl remote-cddl">
+      browser.Close = (
+        method: "browser.closePartition",
+        params: {
+          partition: browser.Partition
+        },
+      )
+      </pre>
+   </dd>
+   <dt>Return Type</dt>
+   <dd>
+    <pre class="cddl local-cddl">
+      browser.closePartitionResult = {
+        partition: browser.Partition
+      }
+    </pre>
+   </dd>
+</dl>
+
+<div algorithm="remote end steps for browser.closePartition">
+
+</div>
+
+#### The browser.getPartitions Command #### {#command-browser-getPartitions}
+
+The <dfn export for=commands>browser.getPartitions</dfn> command returns a list of existing partitions in the browser instance.
```

Comment:
Do you want getPartitions() to be able to return other existing profiles that the user might have created, or only ones created by WebDriver commands? I see both "browser process" and "browser instance" here, but no definitions for them, which might clarify this.

Reply:
That part I have not clarified yet. I think it probably makes sense to only see partitions/profiles/UAs created by the WebDriver session.

Reply:
Historically implementations have differed here; it's possible to start GeckoDriver against a pre-existing profile, whereas SafariDriver always creates a temporary automation session which ins't persisted. I expect implementations will continue to differ, so it should maybe be implementation defined.

Reply:
I don't think "created by the WebDriver session" is the right line, but something like "which are visible to the WebDriver session" or something. Basically, if you can subscribe to events from a specific browsing context, you should be able to see the user context that's part of.

I think that's basically equivalent to making it implementation-defined, but I'm also not sure we need any special language for this feature; the constraint is just that the model has to be consistent, and it's already defacto possible for implementations to have parts of the browser that are not visible to WebDriver clients, but that can be understood at the session level.